### ISSUE TITLE
[bug][clang-tidy][21.1.8] A false positive case of readability-non-const-parameter

### ISSUE THREAD

**Issue by yaoxinliu** [llvm/llvm-project#176623]
```cpp
template<bool>
class A final {
    char* sz_ = {};

public:
    explicit A(char* sz) noexcept : sz_(sz) {}

    void f() { sz_ = {}; }
};

class B final {
    char* sz_ = {};

public:
    explicit B(char* sz) noexcept : sz_(sz) {}

    void f() { sz_ = {}; }
};

int main() {
    // clang-tidy warning of readability-non-const-parameter: Pointer parameter 'p' can be pointer to const. 
    auto const _ = []<bool tc>(char* p) { auto _ = A<tc>(p); }; 

    auto const _ = []<bool tc>(char* p) { auto _ = B(p); }; // ok, no clang-tidy warning as expected. 
}
```

The clang-tidy warning is false positive.

**Comment by SagarJadhav30** [comment]
clang-tidy emits a false positive readability-non-const-parameter warning when a lambda parameter is forwarded to a templated class, but does not emit the warning for an equivalent non-templated class.
